### ISSUE TITLE
[1.328] Can't selecting brushes after play mode (Unity 5.4.2)

### ISSUE THREAD

**Issue by Hisashimaru** [LogicalError/realtime-CSG-for-unity#60]
RealtimeCSG 1.312
Unity 5.4.2f1

Reproduction steps: 
1.Open new scene
2.Make a brush
3.Click play button and then click stop button
4.Select the brush

I got error messages, when I made a brush after play mode.

Error message
```
NullReferenceException: Object reference not set to an instance of an object
RealtimeCSG.CSGSceneManager.GetParentData (RealtimeCSG.ChildNodeData childNode) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Managers/CSGSceneManager.cs:270)
RealtimeCSG.CSGSceneManager.RegisterBrush (RealtimeCSG.CSGBrush brush) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Managers/CSGSceneManager.cs:1076)
RealtimeCSG.CSGSceneManager.OnHierarchyModified () (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Managers/CSGSceneManager.cs:2587)
RealtimeCSG.CSGSceneManager.Refresh (Boolean forceHierarchyUpdate) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Managers/CSGSceneManager.cs:2369)
RealtimeCSG.BaseShapeGenerator.UpdateExtrudedShape (Single height, Boolean registerUndo) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/Generators/GeneratorBase.cs:659)
RealtimeCSG.BaseShapeGenerator.HandleHeightHandles () (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/Generators/GeneratorBase.cs:1081)
RealtimeCSG.BoxGenerator.HandleEditVerticesEvents () (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/Generators/BoxGenerator.cs:827)
RealtimeCSG.BaseShapeGenerator.HandleEvents () (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/Generators/GeneratorBase.cs:818)
RealtimeCSG.BoxGenerator.RealtimeCSG.IBrushGenerator.HandleEvents ()
RealtimeCSG.GenerateBrushTool.HandleEvents () (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/GenerateBrushTool.cs:185)
RealtimeCSG.CSGBrushEditorManager.OnSceneGUI (UnityEditor.SceneView sceneView) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushEditorManager/CSGBrushEditorManager.cs:631)
RealtimeCSG.UnitySceneEventHandler.OnScene (UnityEditor.SceneView sceneView) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Managers/UnitySceneEventHandler.cs:860)
UnityEditor.SceneView.CallOnSceneGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:2046)
UnityEditor.SceneView.HandleSelectionAndOnSceneGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:1381)
UnityEditor.SceneView.OnGUI () (at C:/buildslave/unity/build/Editor/Mono/SceneView/SceneView.cs:1218)
System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:222)
```

Warning message
```
GetModelByIndex(modelIndex) == nullptr

bool __cdecl SetBrushMesh(const int,int,const struct Vector3 *const ,int,const int *const ,const int *const ,int,const struct RealtimeCSG::PolygonInput *const ) (at C:\Projects\Prenominal\UnityCSG\EditorPlugin\unity\unity_csg_brush.cpp:493)
UnityEngine.Debug:LogWarning(Object, Object)
RealtimeCSG.CSGBindings:<RegisterUnityMethods>m__2(String, Int32) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Bindings/CSGBindings.cs:73)
RealtimeCSG.CSGBindings:SetBrushMesh(Int32, Int32, Vector3[], Int32, Int32[], Int32[], Int32, PolygonInput[])
RealtimeCSG.CSGSceneManager:SetBrushMesh(CSGBrush) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Managers/CSGSceneManager.cs:1223)
RealtimeCSG.CSGSceneManager:CheckSurfaceModifications(CSGBrush, Boolean) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Managers/CSGSceneManager.cs:2101)
RealtimeCSG.CSGSceneManager:ValidateBrush(CSGBrush, Boolean) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Managers/CSGSceneManager.cs:2144)
RealtimeCSG.BoxGenerator:CreateControlMeshForBrushIndex(CSGModel, CSGBrush, ShapePolygon, Single, ControlMesh&, Shape&) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/Generators/BoxGenerator.cs:266)
RealtimeCSG.BaseShapeGenerator:UpdateExtrudedShape(Single, Boolean) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/Generators/GeneratorBase.cs:606)
RealtimeCSG.BaseShapeGenerator:HandleHeightHandles() (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/Generators/GeneratorBase.cs:1081)
RealtimeCSG.BoxGenerator:HandleEditVerticesEvents() (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/Generators/BoxGenerator.cs:827)
RealtimeCSG.BaseShapeGenerator:HandleEvents() (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/Generators/GeneratorBase.cs:818)
RealtimeCSG.BoxGenerator:RealtimeCSG.IBrushGenerator.HandleEvents()
RealtimeCSG.GenerateBrushTool:HandleEvents() (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushTools/GenerateBrushTool.cs:185)
RealtimeCSG.CSGBrushEditorManager:OnSceneGUI(SceneView) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/EditorWindows/BrushEditorManager/CSGBrushEditorManager.cs:631)
RealtimeCSG.UnitySceneEventHandler:OnScene(SceneView) (at Assets/Plugins/RealtimeCSG/Editor/Scripts/Managers/UnitySceneEventHandler.cs:860)
UnityEditor.DockArea:OnGUI()
```

**Comment by LogicalError** [comment]
Oh that's weird, I'll investigate

**Comment by LogicalError** [comment]
I haven't been able to reproduce this .. hmm
Is there any additional information you can give me?
Does this happen to you every time? Or just that once? Or sometimes?

**Comment by Hisashimaru** [comment]
It's strange.. the problem happens every time. I have updated to the latest version of RealtimeCSG 1.314, but still happening it.

I record the video.
https://youtu.be/9r4E1TOO_2Y

**Comment by LogicalError** [comment]
Hmm .. ok, that's odd
so ... in this project, you haven't installed any other plugins besides realtime-CSG?

**Comment by Hisashimaru** [comment]
My project installed only realtimeCSG, it's very pure.
I found another information. I can select a brush When I click the brush immediately after back from play mode. Conversely, if I wait about 1~2 seconds and click a brush after back from play mode, I can't select the brush.

**Comment by LogicalError** [comment]
Ok, I still can't repeat it, even after waiting a bit after pressing play. :-/
Can you try one thing? Could you delete your plugins folder in your project (while Unity is closed) and reinstall Realtime-CSG?

**Comment by Hisashimaru** [comment]
I tried reinstall some version of RealtimeCSG with your method, and I noticed new fact. The problem happens version 1.308 or later. I hope this information help you :)

**Comment by LogicalError** [comment]
I managed to reproduce this in Unity 5.4.2 .. problem doesn't exist in 5.3.5 apparently?

**Comment by LogicalError** [comment]
related to #96 and #109

**Comment by LogicalError** [comment]
This is fixed in 1.328! 
You can download it at the update site https://prenominal.nl/updates/
Once this version is available from the asset store, I'll close this ticket
